artmanflow/steps/java_sources_staging_guest.py

```diff
@@ -0,0 +1,158 @@
+                staging_path)
+            if not os.path.exists(dest_path):
+                os.makedirs(dest_path)
+            dests.append(dest)
+
+        unzip_dest = self._guest.guest_root_subpath([staging_name, 'generated'])
+        self.run_command(
+            ['tar', '-pxzf', self._config['generator_artifacts']['sources_zip'],
+             '-C', unzip_dest, '--wildcards', './java/*'],
+            self._guest.guest_root_path())
+        for dest in dests:
+            self.run_command(['git', 'add', dest], staging_path)
+
+    def _git_commit_and_push(self, staging_name):
+        staging_path = self._guest.guest_root_subpath(staging_name)
+        self.run_command(['git', 'status'],
+                         self._guest.guest_root_subpath(staging_name))
+        self.run_command(['git', 'commit', '--allow-empty', '-m',
+                          'Regenerate Java client sources'],
+                         staging_path)
+        self.run_command(['git', 'push', '-u', 'origin',
+                          self._config['staging']['git_branch']],
+                         staging_path)
+
+    def _build_and_test(self, staging_name):
+        if not self._config['staging']['run_tests']:
+            return
+        cwd = self._guest.guest_root_subpath(
+            [staging_name, 'generated', 'java'])
+        self.run_command(['./gradlew', 'clean', 'test'], cwd)
+
+    def _post_pr(self):
+        config = self._config['staging']
+        repo_name = config['git_repo']
+        repo_owner, repo_name = GitUtils.repo_properties(repo_name)
+
+        gh = github3.login(config['git_user_name'],
+                           config['git_security_token'])
+        repo = gh.repository(repo_owner, repo_name)
+
+        pr_base = 'master'
+        # pr_head = "%s:%s" % (config['git_user_name'], config['git_branch'])
+        pr_head = config['git_branch']
+
+        self.puts(
+            'Creating pull request, base: %s, head: %s' % (pr_base, pr_head))
+        pr = repo.create_pull(
+            base=pr_base,
+            body='This PR is automatically generated by Artmanflow tool',
+            head=pr_head,
+            title='Artman Workflow Java sources staging'
+        )
+
+        if not pr:
+            raise RuntimeError('Pull request creation failed.')
+
+        self.puts('Pull request created successfully: %s' % pr.html_url)
+
+        return pr.html_url
+
+    def _dump_output(self, output_yaml):
+        output_yaml_path = self._guest.guest_output_dir_subpath(
+            ConfigUtils.artifact_yaml_name())
+        ConfigUtils.dump_config(output_yaml, output_yaml_path)
+        self.change_file_permissions(output_yaml_path)
+
+    # TODO: this is a crime, do it properly in toolkit instead
+    def _fix_staging_versions(self, staging_name):
+        config = self._config['staging']
+        cwd = self._guest.guest_root_subpath(
+            [staging_name, 'generated', 'java'])
+
+        gax_ver_sed_arg = \
```

Comment:
This would be a lot easier to understand if it was done in native Python and not `sed`.

Reply:
This hack is removed, there is no `sed` anymore (fixed properly on toolkit side)